src/darwin/Framework/CHIP/CHIPDevice.mm

```diff
@@ -1182,23 +1226,29 @@ - (void)subscribeAttributeWithEndpointId:(NSNumber * _Nullable)endpointId
 
         CHIPReadClientContainer * container = [[CHIPReadClientContainer alloc] init];
         container.deviceId = self.cppDevice->GetDeviceId();
-        container.pathParams = Platform::New<app::AttributePathParams>();
-        if (endpointId) {
-            container.pathParams->mEndpointId = static_cast<chip::EndpointId>([endpointId unsignedShortValue]);
-        }
-        if (clusterId) {
-            container.pathParams->mClusterId = static_cast<chip::ClusterId>([clusterId unsignedLongValue]);
-        }
-        if (attributeId) {
-            container.pathParams->mAttributeId = static_cast<chip::AttributeId>([attributeId unsignedLongValue]);
+
+        chip::app::AttributePathParams attributePathParams[kMaxAllowedPaths];
```

Comment:
But then the ReadClient will have a pointer to this stack object, and if it tries to resubscribe things will blow up, no?